src/firefly/java/edu/caltech/ipac/astro/ibe/datasource/AtlasIbeDataSource.java

```diff
@@ -0,0 +1,395 @@
+                table = dsInfo.get(TABLE_KEY);
+            } else {
+                if (!StringUtils.isEmpty(dsInfo.get(DS_KEY))) {
+                    String[] defs = dsInfo.get(DS_KEY).split("\\.");
+                    if (defs.length == 2) {
+                        schema = defs[0];
+                        table = defs[1];
+                    }
+                }
+
+            }
+        }
+        setupDS(null, schema, table);
+    }
+
+    private void setupDS(String ibeHost, String dataset, String table) {
+
+        if (StringUtils.isEmpty(ibeHost)) {
+            ibeHost = AppProperties.getProperty("atlas.ibe.host", "https://irsa.ipac.caltech.edu");
+        }
+        setIbeHost(ibeHost);
+        setMission(ATLAS);
+        setDataset(dataset);
+        setTableName(table);
+    }
+
+    public DS getDS() {
+        return ds;
+    }
+
+    @Override
+    public IbeDataParam makeDataParam(Map<String, String> pathInfo) {
+        IbeDataParam dataParam = new IbeDataParam();
+
+        String fname = pathInfo.get("fname");
+
+        String root = getImageRoot();
+
+        dataParam.setFilePath(root + "/" + fname);
+
+        // check cutout params
+        // look for ra_obj first - moving object search
+        String subLon = pathInfo.get("ra_obj");
+        if (StringUtils.isEmpty(subLon)) {
+            // next look for in_ra (IBE returns this)
+            subLon = pathInfo.get("in_ra");
+            if (StringUtils.isEmpty(subLon)) {
+                // all else fails, try using crval1
+                subLon = pathInfo.get("crval1");
+            }
+        }
+
+        // look for dec_obj first - moving object search
+        String subLat = pathInfo.get("dec_obj");
+        if (StringUtils.isEmpty(subLat)) {
+            // next look for in_dec (IBE returns this)
+            subLat = pathInfo.get("in_dec");
+            if (StringUtils.isEmpty(subLat)) {
+                // all else fails, try using crval2
+                subLat = pathInfo.get("crval2");
+            }
+        }
+        String subSize = pathInfo.get("subsize");
+
+        if (!StringUtils.isEmpty(subLon) && !StringUtils.isEmpty(subLat) && !StringUtils.isEmpty(subSize)) {
+            dataParam.setCutout(true, subLon + "," + subLat, subSize);
+        }
+
+        if (dataParam.getFilePath() != null && dataParam.getFilePath().endsWith(".gz")) {
+            dataParam.setDoZip(true);
+        }
+
+        return dataParam;
+    }
+
+    @Override
+    public IbeQueryParam makeQueryParam(Map<String, String> queryInfo) {
+
+        // source search
+        IbeQueryParam queryParam = new IbeQueryParam();
+
+        // process POS - target search
+        String userTargetWorldPt = queryInfo.get("UserTargetWorldPt");
+        if (userTargetWorldPt != null) {
+            WorldPt pt = WorldPt.parse(userTargetWorldPt);
+            if (pt != null) {
+                pt = Plot.convert(pt, CoordinateSys.EQ_J2000);
```

Comment:
glad you remembered to do the convert.
it could be simplified to
```
 WorldPt pt = Plot.convert(WorldPt.parse(userTargetWorldPt));
```
It will not throw an exception, just return a null.

If you want to take out another if block. then:
```
WorldPt pt= Plot.convert(
      WorldPt.parse( queryInfo.get("UserTargetWorldPt")));
if (pt!=null) {
     ....
}
```